client/domain/ReinforcementObjectFactory.js

```diff
@@ -59,20 +59,24 @@ tie.factory('ReinforcementObjectFactory', [
       return this._task;
     };
 
-    Reinforcement.prototype.addToPassedList = function(key, value) {
-      this._passedList[key] = value;
+    Reinforcement.prototype.addToPassedTagsList = function(key, value) {
+      this._passedTagsList[key] = value;
     };
 
-    Reinforcement.prototype.addToPastFailsList = function(key, value) {
-      this._pastFailsList[key] = value;
+    Reinforcement.prototype.addToPastFailedCasesList = function(key, value) {
+      this._pastFailedCasesList[key] = value;
     };
 
-    Reinforcement.prototype.getPassedList = function() {
-      return this._passedList;
+    Reinforcement.prototype.getPassedTagsList = function() {
+      return this._passedTagsList;
     };
 
-    Reinforcement.prototype.getPastFailsList = function() {
-      return this._pastFailsList;
+    Reinforcement.prototype.getPastFailedCasesList = function() {
+      return this._pastFailedCasesList;
+    };
+
+    Reinforcement.prototype.hasPastFailedCase = function(caseInput) {
+       return (caseInput in this._pastFailedCasesList);
```

Comment:
return this._pastFailedCasesList.hasOwnProperty(caseInput)

Btw, are the caseInput's guaranteed to be strings / hashable?